recipe_scrapers/quitoque.py

```diff
@@ -0,0 +1,95 @@
+# mypy: allow-untyped-defs
+
+import requests
+
+from ._abstract import AbstractScraper
+
+
+class QuiToque(AbstractScraper):
+    def __init__(self, url, proxies=None, timeout=None, *args, **kwargs):
+        super().__init__(url=url, proxies=proxies, timeout=timeout, *args, **kwargs)
+        recipe_id = url.split("/")[4]
+        data_url = "https://mgs.quitoque.fr/graphql"
+        data_body = {
+            "operationName": "getRecipe",
+            "variables": {"id": recipe_id},
+            "query": "query getRecipe($id:ID!){\nrecipe(id:$id){\nname\nshortDescription\nfacets{\nname\n}\nimage\nnutritionalInformations{\nkiloCalorie\nfat\nsaturatedFat\ncarbohydrate\nsugarCarbohydrate\nfiber\nprotein\nsalt\n}\nfacets{\nname\n}\npools{\nnbPerson\ncookingModes{\ncookingTime\nsteps{\ndescription\n}\nstacks{\ntools{\nname\n}\ncupboardIngredients{\nquantity\nliteralQuantity\nproduct{\nname\n}\n}\ningredients{\nliteralQuantity\nquantity\nproduct{\nname\n}\n}\n}\nwaitingTime\n}\n}\n}\n}",
+        }
```

Comment:
Perhaps we could extract the query here into a separate variable for readability and maintainability?  Something like:

```py
query = """
query getRecipe($id: ID!) {
  recipe(id: $id) {
    name
      shortDescription
      facets{
        name
      }
    image
      nutritionalInformations{
        kiloCalorie
          fat
          saturatedFat
          carbohydrate
          sugarCarbohydrate
          fiber
          protein
          salt
      }
    facets{
      name
    }
    pools{
      nbPerson
        cookingModes{
          cookingTime
            steps{
              description
            }
          stacks{
            tools{
              name
            }
            cupboardIngredients{
              quantity
                literalQuantity
                product{
                  name
                }
            }
            ingredients{
              literalQuantity
                quantity
                product{
                  name
                }
            }
          }
          waitingTime
        }
    }
  }
}
data_body = {
    ...
    "query": query,
}
"""
```

What does the exclamation mark (`!`) do after the `ID` parameter, by the way?  Is that to make it a required parameter?

Reply:
Also: it looks like `facets{name}` is specified twice in the query - do we need it at all?  (we already retrieve the individual recipe `name`)

Reply:
Sure thing ! For readability, it will be better to make it multiline. 

The exclamation mark indicate that the variable `$id` is of type `ID!` in our query. Variable that is defined into the body referencing the id of the recipe.

My bad, query referencing twice the `facets{ name }` is a mistake from me.  We use `facets` for categories.

I correct those.
Thanks again.